Dockerfile

```diff
@@ -1,3 +1,16 @@
-FROM starefossen/github-pages:onbuild
+# Get Jekyll build env and Nginx env
+FROM docs/docker.github.io:docs-builder AS builder
 
-CMD jekyll serve -d /_site --watch -H 0.0.0.0 -P 4000
+# Build the docs from this branch
+COPY . /source
+RUN jekyll build --source /source --destination /site
```

Comment:
Given that this branch only contains HTML, should we skip these steps entirely, or is Jekyll still doing some templating? (looks like we still need the `docs-builder` image for the nginx-config though 😞 (so perhaps a "config" image would be good after all).

Note that `COPY --from=<image>:<tag>` is also possible, and will pull the image you're copying from 🎉 

```Dockerfile
FROM nginx:alpine

# Copy the Nginx config from the docs-builder image
COPY --from=docs/docker.github.io:docs-builder /conf/nginx-overrides.conf /etc/nginx/conf.d/default.conf

# Copy the HTML to the default location for Nginx and serve it
COPY . /usr/share/nginx/html
CMD echo -e "Docker docs are viewable at:\nhttp://0.0.0.0:4000"; exec nginx -g 'daemon off;'
```

Reply:
So difference between using Jekyll first or directly copying is;

```
$ docker build -t docs/docker.github.io:v1.4 --target=builder . 
$ docker run --rm  docs/docker.github.io:v1.4 diff --brief -Nr /source /site

Files /source/.dockerignore and /site/.dockerignore differ
Files /source/.gitignore and /site/.gitignore differ
Files /source/_config.yml and /site/_config.yml differ
Files /source/articles/basics.md~ and /site/articles/basics.md~ differ
Files /source/faq.md~ and /site/faq.md~ differ
Files /source/hooks/post_push and /site/hooks/post_push differ
Files /source/installation/images/win/_01.gif and /site/installation/images/win/_01.gif differ
Files /source/installation/images/win/_02.gif and /site/installation/images/win/_02.gif differ
Files /source/installation/images/win/_06.gif and /site/installation/images/win/_06.gif differ
Files /source/installation/mac.md~ and /site/installation/mac.md~ differ
Files /source/installation/windows.md~ and /site/installation/windows.md~ differ
Files /source/reference/api/_static/io_oauth_authorization_page.png and /site/reference/api/_static/io_oauth_authorization_page.png differ
Files /source/reference/commandline/cli.md~ and /site/reference/commandline/cli.md~ differ
Files /source/userguide/dockerizing.md~ and /site/userguide/dockerizing.md~ differ
```

Opened https://github.com/docker/docker.github.io/pull/5382 to get rid of those tmp-files and some unused images